libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshControllerTest.kt

```
@@ -1,307 +1,600 @@
+                Month.MAY,
+                22,
+                13,
+                0,
+                0
+            )
+            val currentRoute = createTestTwoLegRoute(
+                firstLegIncidents = listOf(
+                    createIncident(
+                        id = "1",
+                        endTime = "2022-05-22T10:00:00Z",
+                    ),
+                ),
+                secondLegIncidents = listOf(
+                    createIncident(
+                        id = "2",
+                        endTime = "2022-05-22T10:00:00Z"
+                    ),
+                    createIncident(
+                        id = "3",
+                        endTime = "2022-05-22T13:00:01Z"
+                    ),
+                )
+            )
+            val currentLegIndexProvider = { 1 }
+            val directionsSession = mockk<DirectionsSession>(relaxed = true) {
+                var refreshAttempt = 0L
+                every { requestRouteRefresh(any(), any(), any()) } answers {
+                    refreshAttempt++
+                    refreshAttempt
+                }
+            }
+            val routeRefreshOptions = RouteRefreshOptions.Builder()
+                .intervalMillis(30_000L)
+                .build()
+            val routeRefreshController = createRouteRefreshController(
+                routeRefreshOptions = routeRefreshOptions,
+                routeDiffProvider = DirectionsRouteDiffProvider(),
+                currentLegIndexProvider = currentLegIndexProvider,
+                currentDateTimeProvider = { currentTime },
+                directionsSession = directionsSession
+            )
+            // act
+            val refreshedRoutesDeferred = async {
+                routeRefreshController.refresh(listOf(currentRoute))
+            }
+            // in case of timeout controller will wait for the third response, so congestion will be
+            // invalidated after 4X refresh intervals
```

If we don't want to expose configuration parameter, for me, attempt based cleanup makes more sense. But it increases time if router doesn't respond. Or cleanup too fast if router fails fast.

---

> But it increases time if router doesn't respond

This makes sense :+1: 

> Or cleanup too fast if router fails fast

How can this happen? It's always going to be at minimum `3 * interval`, isn't it?

---

> It's always going to be at minimum 3 * interval, isn't it?

Yes.

> Or cleanup too fast if router fails fast.

>  How can this happen?

I forgot to write that I was considering cleaning current route after 2 attempts 😅  , so that in case of no internet connection, which causes no response in our router, route cleaned up after 3 * interval time. But cleanup will be too fast if router fails fast. Never mind if you agree with current approach.